s2a/src/main/java/io/grpc/s2a/internal/handshaker/S2AProtocolNegotiatorFactory.java

```diff
@@ -233,6 +254,9 @@ public void onSuccess(SslContext sslContext) {
                           SharedResourcePool.forResource(GrpcUtil.SHARED_CHANNEL_EXECUTOR))
                       .newHandler(grpcHandler);
 
+              ctx.pipeline().addAfter(ctx.name(), /* name= */ null,
+                  new S2AStubCleanupNegotiationHandler(handler,
+                    grpcHandler.getNegotiationLogger(), s2aStub));
               // Remove the bufferReads handler and delegate the rest of the handshake to the TLS
```

Comment:
The comment doesn't seem to match the code since the event gets fired before the bufferReads handler is removed while the comment seems to me to be in the other order.

Reply:
Done in https://github.com/grpc/grpc-java/pull/11600/commits/a7fd45f77f1cb93baa6a47d6e845e8cc0fecdd40